server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -303,7 +293,11 @@ private void doComplete(boolean shouldRetry) {
             }
         } catch (Throwable t) {
             logger.error("Exception in RemoteStoreRefreshListener.afterRefresh()", t);
+        } finally {
+            // Update the segment tracker with the final upload status as seen at the end
+            updateFinalUploadStatusInSegmentTracker(shouldRetry.get() == false, bytesBeforeUpload, startTimeInNS);
```

Comment:
> We should rename this to something like updateSegmentTracker because it needs to be invoked even if there's an error before upload.

We are renaming some of these methods in coming PRs as part of segments download & translog stat upload/download integration with stats api.

>Do we need a finally? We are anyways catching a throwable.

Have addressed this already.